### PR TITLE
Use X11's find module

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

[These targets](https://cmake.org/cmake/help/v3.16/module/FindX11.html) were added in CMake 3.14 so this wasn't accessible in prior versions of SFML. Using find_package and find_dependency let us express this dependency in a much more succinct way.

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by eXpl0it3r — APPROVED** [review]

File: src/SFML/Window/CMakeLists.txt [hunk 1]
```diff
@@ -285,8 +285,8 @@ if(SFML_OS_LINUX OR SFML_OS_FREEBSD OR SFML_OS_OPENBSD OR SFML_OS_NETBSD)
         sfml_find_package(GBM INCLUDE "GBM_INCLUDE_DIR" LINK "GBM_LIBRARY")
         target_link_libraries(sfml-window PRIVATE drm gbm EGL)
     else()
-        sfml_find_package(X11 INCLUDE "X11_INCLUDE_DIR" LINK "X11_X11_LIB" "X11_Xrandr_LIB" "X11_Xcursor_LIB")
-        target_link_libraries(sfml-window PRIVATE X11)
+        find_package(X11 REQUIRED)
```
Comment (kimci86) [inline src/SFML/Window/CMakeLists.txt:288]:
```suggestion
        find_package(X11 REQUIRED COMPONENTS Xcursor Xrandr)
```

File: cmake/SFMLConfigDependencies.cmake.in [hunk 2]
```diff
@@ -40,9 +42,7 @@ if(SFML_STATIC_LIBRARIES)
     list(FIND SFML_FIND_COMPONENTS "Window" FIND_SFML_WINDOW_COMPONENT_INDEX)
     if(FIND_SFML_WINDOW_COMPONENT_INDEX GREATER -1)
         if(FIND_SFML_OS_LINUX OR FIND_SFML_OS_FREEBSD)
-            sfml_bind_dependency(TARGET X11 FRIENDLY_NAME "X11" SEARCH_NAMES "X11")
-            sfml_bind_dependency(TARGET X11 FRIENDLY_NAME "Xrandr" SEARCH_NAMES "Xrandr")
-            sfml_bind_dependency(TARGET X11 FRIENDLY_NAME "Xcursor" SEARCH_NAMES "Xcursor")
+            find_dependency(X11)
```
Comment (kimci86) [inline cmake/SFMLConfigDependencies.cmake.in:45]:
```suggestion
            find_dependency(X11 REQUIRED COMPONENTS Xcursor Xrandr)
```

**Review by kimci86 — COMMENTED** [review]
I am late, but I think we can and should be as explicit as before when listing dependencies. See my suggested changes.

File: src/SFML/Window/CMakeLists.txt (hunk 1 above)
Comment (ChrisThrasher) [reply src/SFML/Window/CMakeLists.txt:288]: Is it possible that the X11 package may not contain these components or is this mostly about expressiveness? My impression of [FindX11.cmake](https://cmake.org/cmake/help/latest/module/FindX11.html) was that all targets were always available.

Comment (kimci86) [reply src/SFML/Window/CMakeLists.txt:288]: Not all targets are always available. For example, some need a specific apt package (see [ci.yml](https://github.com/SFML/SFML/blob/eb162df723c178c330f7ec33dc6a288267f2afe3/.github/workflows/ci.yml#L60)). I tested configuring on my system and a great proportion of cache variables from FindX11.cmake end with NOTFOUND.

Comment (ChrisThrasher) [reply src/SFML/Window/CMakeLists.txt:288]: In that case I'm cool with us explicitly listing components.

**Pull request merged**